src/extension/importer/core/src/main/java/org/geoserver/importer/ImportContext.java

```diff
@@ -247,7 +252,7 @@ public ImportTask task(long id) {
         return defaultTransforms;
     }
 
-    private void updateState() {
+    private void updateState() throws IOException {
```

Comment:
Ok, that does seem severe enough to merit propagating the exception up to the user. My only remaining comment would be for that the UI should properly report the exception as an error message, rather than throwing a RuntimeException and causing a stacktrace on the page.

Reply:
@tbarsballe done. Thanks for your review!